### PR TITLE
Add type annotations to Series 

### PR DESCRIPTION
**Pull request by xinrong-databricks**
Add type annotations (mostly return types) to functions in Series.py. This helps autocompletion in various editors.

### CODE REVIEW

File: databricks/koalas/series.py [hunk 1]
```diff
@@ -778,7 +780,7 @@ def ne(self, other):
         """
         return self != other
 
-    def divmod(self, other):
+    def divmod(self, other) -> Tuple[Any, Any]:
```
Comment (ueshin) [inline databricks/koalas/series.py:783]: `Tuple["Series", "Series"]`?

File: databricks/koalas/series.py [hunk 2]
```diff
@@ -798,7 +800,7 @@ def divmod(self, other):
         """
         return (self.floordiv(other), self.mod(other))
 
-    def rdivmod(self, other):
+    def rdivmod(self, other) -> Tuple[Any, Any]:
```
Comment (ueshin) [inline databricks/koalas/series.py:803]: ditto.

File: databricks/koalas/series.py [hunk 3]
```diff
@@ -1547,7 +1552,7 @@ def toPandas(self) -> pd.Series:
 
     toPandas.__doc__ = to_pandas.__doc__
 
-    def to_list(self):
+    def to_list(self) -> list:
```
Comment (ueshin) [inline databricks/koalas/series.py:1555]: `List`?

File: databricks/koalas/series.py [hunk 4]
```diff
@@ -1445,7 +1450,7 @@ def to_clipboard(self, excel=True, sep=None, **kwargs):
 
     to_clipboard.__doc__ = DataFrame.to_clipboard.__doc__
 
-    def to_dict(self, into=dict):
+    def to_dict(self, into=dict) -> Mapping:
```
Comment (HyukjinKwon) [inline databricks/koalas/series.py:1453]: nit: `typing.Dict`?

File: databricks/koalas/series.py [hunk 5]
```diff
@@ -312,6 +312,8 @@
 # Needed to disambiguate Series.str and str type
 str_type = str
 
+Scalar = Union[int, float, bool, str, bytes, complex, None]
```
[the remark this answers is not in the document]
Comment (xinrong-databricks) [reply databricks/koalas/series.py]: Makes sense! Moved `Scalar` to `typehints.py`.
I exclude numpy types for now. I think it's a great idea to revisit them later.

File: databricks/koalas/series.py (hunk 1 above)
Comment (xinrong-databricks) [reply databricks/koalas/series.py]: Modified!

File: databricks/koalas/series.py [hunk 6]
```diff
@@ -1445,7 +1448,7 @@ def to_clipboard(self, excel=True, sep=None, **kwargs):
 
     to_clipboard.__doc__ = DataFrame.to_clipboard.__doc__
 
-    def to_dict(self, into=dict):
+    def to_dict(self, into=dict) -> Dict:
```
Comment (HyukjinKwon) [inline databricks/koalas/series.py:1451]: Oops @xinrong-databricks, sorry I think what you did with `collections.abc.Mapping` is more correct. Looks like `info` might not necessarily be a dict.

File: databricks/koalas/series.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (ueshin) [reply databricks/koalas/series.py]: cc @itholic

File: databricks/koalas/series.py (hunk 6 above)
Comment (xinrong-databricks) [reply databricks/koalas/series.py]: That's totally fine. 😄  Thank you!

File: databricks/koalas/series.py [hunk 7]
```diff
@@ -3975,7 +3984,7 @@ def mode(self, dropna=True) -> "Series":
 
         return first_series(DataFrame(internal))
 
-    def keys(self):
+    def keys(self) -> Union["ks.Index", "ks.MultiIndex"]:
```
[the remark this answers is not in the document]
Comment (xinrong-databricks) [reply databricks/koalas/series.py]: Modified!

**Review by ueshin — APPROVED** [review]
LGTM.

**Comment by ueshin** [comment]
Thanks! merging.

**Comment by xinrong-databricks** [comment]
@ueshin Thank you!

**Pull request merged**